frontend/src/pages/calculator/index.js

```diff
@@ -1,13 +1,247 @@
-import './CalculatorElements';
+import axios from 'axios';
+import { useNavigate } from 'react-router-dom';
+import 'antd/dist/antd.css';
+import { Radio, Space } from 'antd';
+import {
+  FormContainer,
+  CustomSlider,
+  CustomInput,
+  BoxOne,
+  BoxTwo,
+  BoxThree,
+  BoxFour,
+  BoxFive,
+  BoxSix,
+  BoxSeven,
+  ButtonContainer,
+  CalculatorSubmitButton,
+  IllustrationContainer,
+} from './CalculatorElements';
+import Footer from '../../components/footerPurple';
+import { SectionTitle } from '../../components/sectionTitle/SectionTitle';
+import calculatorIllustration from '../../images/calculatorIllustration.svg';
 
+const formItemLayout = {
+  labelCol: { span: 24 },
+  wrapperCol: { span: 24 },
+};
 function Calculator() {
+  const [creditValue, setCreditValue] = React.useState(0);
+  const [downPaymentValue, setDownPaymentValue] = React.useState(0);
+  const [loanTermValue, setLoanTermValue] = React.useState(0);
+  const [interestRateValue, setInterestRateValue] = React.useState(0);
+  const [commissionValue, setComissionValue] = React.useState(0);
+  const [interestTypeValue, setInterestTypeValue] = React.useState(1);
+  const [installmentTypeValue, setInstallmentType] = React.useState(1);
+  const navigate = useNavigate();
+
+  const onCreditValueChange = (value) => {
+    setCreditValue(value);
+  };
+  const onDownPaymentValueChange = (value) => {
+    setDownPaymentValue(value);
+  };
+  const onLoanTermValueChange = (value) => {
+    setLoanTermValue(value);
+  };
+  const onSetInterestRateValueChange = (value) => {
+    setInterestRateValue(value);
+  };
+  const onComissionValueChange = (value) => {
+    setComissionValue(value);
+  };
+
+  const onInterestTypeValueChange = (value) => {
+    setInterestTypeValue(value.target.value);
+  };
+
+  const onInstallmentTypeValueChange = (value) => {
+    setInstallmentType(value.target.value);
+  };
+
+  const formatter = (value) => value;
+
   return (
-    <PlaceholderStyle>
-      <h1>Placeholder for calculator</h1>
-    </PlaceholderStyle>
+    <>
+      <FormContainer
+        {...formItemLayout}
+        onFinish={(values) => {
+          axios
+            .put(
+              'http://127.0.0.1:5000/mortgageCalculator',
+              JSON.stringify(values),
+              {
+                headers: {
+                  'Content-Type': 'application/json',
+                },
+              }
+            )
+            .then((response) =>
+              navigate('/wyniki-obliczen', {
+                state: { response: response.data },
+              })
+            )
+            .catch((error) => {
+              console.log(error);
+            });
+        }}
+      >
+        <IllustrationContainer>
+          <img
+            className="img"
+            src={calculatorIllustration}
+            alt="calculator illustration"
+          />
+        </IllustrationContainer>
+        <BoxOne>
```

Comment:
Do you need creating components like `BoxOne`, `BoxTwo`, etc.? I think you could create models for each box with configured properties, pass them to the function that returns JSX with whole box.

Consinder an example:

```jsx
const formModels = [
  { onChangeHandler: onInterestTypeValueChange, name: "interest_type", ... },
  { onChangeHandler: onInstallmentTypeValueChange, name: "installment_type", ... },
  ...
];

const renderBox = ({name, title, onChangeHandler, ...}) => (
        <Box>
          <SectionTitle>{title}</SectionTitle>
          <FormContainer.Item name={name}>
            <Radio.Group
              onChange={onChangeHandler}
              value={value}
            >
              <Space direction="vertical">
                <Radio value="descending">{firstRadioText}</Radio>
                <Radio value="fixed">{secondRadioText}</Radio>
              </Space>
            </Radio.Group>
          </FormContainer.Item>
        </Box>
);
```

...and inside JSX:

```jsx
{formModels.map(renderBox)}
```

You have two types of boxes (one with radios and one with sliders, so it could be also parametrized with something like `boxType`. Then you can return the right type inside `renderBox` function using simple `if` statement or even `switch`.

Reply:
solved